Source/NetworkClients/Stubs/WeatherClientStub.swift

```diff
@@ -0,0 +1,47 @@
+//  Copyright © 2016 Gavan Chan. All rights reserved.
+
+final class WeatherClientStub: WeatherProviding {
+
+  static let shared: WeatherClientStub = .init()
+
+  fileprivate static let cities: Array<City> = [
+    (7839805, "Melbourne"),
+    (2147714, "Sydney"),
+    (2063523, "Perth"),
+    (2078025, "Adelaide"),
+    (7839562, "Brisbane"),
+    (2163355, "Hobart"),
+    (7839402, "Darwin"),
+  ].map(City.init)
+
+  func weather(for cityID: City.Identifier, in unit: UnitTemperature = .celsius) -> Observable<Weather> {
+    let matchingCityID: ((City) -> Bool) = { $0.identifier == cityID }
+
+    let city: City
+
+    if let _city = WeatherClientStub.cities.first(where: matchingCityID) {
+      city = _city
+    } else {
+      city = City(identifier: 10, name: "Unknown City")
+    }
+
+    return Observable<Weather>.create { observer in
```

Comment:
We can introduce an internal flag here to make sure that the data is not sent if the work has been externally disposed.